src/v/config/rjson_serialization.cc

```diff
@@ -62,6 +61,16 @@ void rjson_serialize(
         w.String((*(v.get_truststore_file())).c_str());
     }
 
+    if (v.get_principal_mapping_rules()) {
+        w.Key("principal_mapping_rules");
+        w.String(*v.get_principal_mapping_rules());
+    }
+}
+
+void rjson_serialize(
+  json::Writer<json::StringBuffer>& w, const config::tls_config& v) {
+    w.StartObject();
+    rjson_serialize_impl(w, v);
     w.EndObject();
 }
```

Comment:
just curious what this json serialization integrates with (e.g. maybe it is pretty printing or integrating with config system), and if we should remove it in the interest of further obfuscation of the feature?

Reply:
I'm not actually sure where it's used. Almost certainly as part of this, if nowhere else: https://github.com/BenPope/redpanda/blob/0b98ddf4ac8a975fd04ba26339304605f3abf2cb/src/v/config/config_store.h#L154-L159

Reply:
I don't know if it's worth obfuscating, `principal_mapping_rules` will only be printed if it has been configured.

Reply:
that's what I thought but wanted to double check